### DISCUSSION TITLE
Periodic boundary conditions do not appear to apply for RDF calculations.

### DISCUSSION THREAD

**Discussion by Ko-o-o** [MDAnalysis/mdanalysis#4644] [Q&A] [answered]
Hi, 

I'm trying calculation of RDF for bulk water system. 
MD simulation was performed by LAMMPS at NVT ensemble.
I used lammps dump file as a trajectory data.
'InterRDF' results did not converge to one, but fell to zero.
I guess the problem caused by periodic boundary description, but I don't know how to solve. 

My python script for RDF calculation is 
[MDA_H2ORDF.txt](https://github.com/user-attachments/files/16424740/MDA_H2ORDF.txt)

and the results of RDF for oxygen-oxygen is
![rdf](https://github.com/user-attachments/assets/cfbb7088-c12e-4501-95d7-bc0839d76919)

Thanks!

**Comment by hmacdope** [comment, marked as answer]
Can you try extending the "Range" argument to say 50? ie [0.1, 50]. It does look like there is an issue though.

**Reply by hmacdope** [reply]
Also what does u.dimensions print?

**Reply by Ko-o-o** [reply]
Thank you for your swift responce.

I attached the result of extending the range up to 100.
![rdf](https://github.com/user-attachments/assets/7cef23f9-887d-4138-97c3-8cbca1aa9249)

u.dimentions printed '[14.4089 14.4089 14.4089 90.     90.     90.    ]'.
O.dimensions returned the same output.

**Reply by hmacdope** [reply]
What is happening is that your RDF has converged to 1  in the < 14 range (ie the size of your box). 

MDAnalysis does correctly take PBCs into account when calculating the RDF but doesn't do PBC extension of the simulation cell to add more copies such that you would see additional density out beyond 14. It also doesn't **automatically** do unwrapping of the trajectory such that the positions of the particles are in not wrapped in space. 

 I think what will fix this is applying an nojump unwrapping transform to your simulation. 
 
See here on how to do that: https://docs.mdanalysis.org/stable/documentation_pages/transformations/nojump.html

**Reply by Ko-o-o** [reply]
Do you mean like this?
[MDA_H2ORDF.txt](https://github.com/user-attachments/files/16449035/MDA_H2ORDF.txt)
This didn't make any changes...

Parhups DUMP file problem?
[lmp_out_reduced.dump.txt](https://github.com/user-attachments/files/16449117/lmp_out_reduced.dump.txt)
